### PR TITLE
update all metrics to include otelcol_ prefix

### LINKED ISSUE #9315: prefix should be consistent for internal metrics
Currently, metrics generated by the collector include the prefix `otelcol` as configured here: https://github.com/open-telemetry/opentelemetry-collector/blob/8131d31601b886ad90b2cc8ae2e83ea96a9e2ca6/service/internal/proctelemetry/config.go#L215

This is not the case when configuring metrics to be emitted via OTLP, as the prefix configuration is specific to prometheus. I'd like to propose the prefix be applied to the metric names, so that they remain consistent when users switch from one mechanism to another.

### PR DESCRIPTION
**Pull request by codeboten**
This ensures the consistency for folks emitting metrics w/ OTLP until the OTEP to specify pipeline telemetry is completed.

Waiting on https://github.com/open-telemetry/opentelemetry-collector/pull/9775 before moving this forward

Fixes #9315

### CODE REVIEW

**Comment by codeboten** [comment]
One thing to note with this change is that metrics that are generated by an instrumentation library or by a component that instantiates its own metric will not be prefixed as the prefixing is now done manually. You can see an example of this in the results below 

Here's the comparison of the metric names for both Prometheus and OTLP emitted metrics before and after this change. 


### Before this change

#### Prometheus metrics

```
# HELP otelcol_exporter_send_failed_spans Number of spans in failed attempts to send to destination.
# TYPE otelcol_exporter_send_failed_spans counter
otelcol_exporter_send_failed_spans{exporter="debug",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 0
# HELP otelcol_exporter_sent_spans Number of spans successfully sent to destination.
# TYPE otelcol_exporter_sent_spans counter
otelcol_exporter_sent_spans{exporter="debug",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 2
# HELP otelcol_process_cpu_seconds Total CPU user and system time in seconds
# TYPE otelcol_process_cpu_seconds counter
otelcol_process_cpu_seconds{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 0.05
# HELP otelcol_process_memory_rss Total physical memory (resident set size)
# TYPE otelcol_process_memory_rss gauge
otelcol_process_memory_rss{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 2.4100864e+07
# HELP otelcol_process_runtime_heap_alloc_bytes Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')
# TYPE otelcol_process_runtime_heap_alloc_bytes gauge
otelcol_process_runtime_heap_alloc_bytes{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 3.010136e+06
# HELP otelcol_process_runtime_total_alloc_bytes Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')
# TYPE otelcol_process_runtime_total_alloc_bytes counter
otelcol_process_runtime_total_alloc_bytes{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 4.247488e+06
# HELP otelcol_process_runtime_total_sys_memory_bytes Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')
# TYPE otelcol_process_runtime_total_sys_memory_bytes gauge
otelcol_process_runtime_total_sys_memory_bytes{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 1.2274704e+07
# HELP otelcol_process_uptime Uptime of the process
# TYPE otelcol_process_uptime counter
otelcol_process_uptime{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 11.321643
# HELP otelcol_processor_batch_batch_send_size Number of units in the batch
# TYPE otelcol_processor_batch_batch_send_size histogram
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="10"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="25"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="50"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="75"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="100"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="250"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="500"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="750"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="1000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="2000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="3000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="4000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="5000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="6000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="7000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="8000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="9000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="10000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="20000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="30000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="50000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="100000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="+Inf"} 1
otelcol_processor_batch_batch_send_size_sum{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 2
otelcol_processor_batch_batch_send_size_count{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 1
# HELP otelcol_processor_batch_metadata_cardinality Number of distinct metadata value combinations being processed
# TYPE otelcol_processor_batch_metadata_cardinality gauge
otelcol_processor_batch_metadata_cardinality{service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 1
# HELP otelcol_processor_batch_timeout_trigger_send Number of times the batch was sent due to a timeout trigger
# TYPE otelcol_processor_batch_timeout_trigger_send counter
otelcol_processor_batch_timeout_trigger_send{processor="batch",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev"} 1
# HELP otelcol_receiver_accepted_spans Number of spans successfully pushed into the pipeline.
# TYPE otelcol_receiver_accepted_spans counter
otelcol_receiver_accepted_spans{receiver="otlp",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",transport="grpc"} 2
# HELP otelcol_receiver_refused_spans Number of spans that could not be pushed into the pipeline.
# TYPE otelcol_receiver_refused_spans counter
otelcol_receiver_refused_spans{receiver="otlp",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",transport="grpc"} 0
# HELP otelcol_rpc_server_duration Measures the duration of inbound RPC.
# TYPE otelcol_rpc_server_duration histogram
otelcol_rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="0"} 0
otelcol_rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",service_instance_id="d64666ef-1bf6-48d1-9cfd-ab242e9f1f50",service_name="otelcorecol",service_version="0.96.0-dev",le="5"} 1
```

#### OTLP metrics

```
        {
          "Name": "processor/batch/metadata_cardinality",
          "Description": "Number of distinct metadata value combinations being processed",
          "Unit": "1",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [],
                "StartTime": "2024-03-15T09:13:05.950473-07:00",
                "Time": "2024-03-15T09:13:09.265554-07:00",
                "Value": 1
              }
            ],
            "Temporality": "CumulativeTemporality",
            "IsMonotonic": false
          }
```

### After this change

#### Prometheus metrics

```
# HELP otelcol_exporter_send_failed_spans Number of spans in failed attempts to send to destination.
# TYPE otelcol_exporter_send_failed_spans counter
otelcol_exporter_send_failed_spans{exporter="debug",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 0
# HELP otelcol_exporter_sent_spans Number of spans successfully sent to destination.
# TYPE otelcol_exporter_sent_spans counter
otelcol_exporter_sent_spans{exporter="debug",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 2
# HELP otelcol_process_cpu_seconds Total CPU user and system time in seconds
# TYPE otelcol_process_cpu_seconds counter
otelcol_process_cpu_seconds{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 0.04
# HELP otelcol_process_memory_rss Total physical memory (resident set size)
# TYPE otelcol_process_memory_rss gauge
otelcol_process_memory_rss{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 2.3904256e+07
# HELP otelcol_process_runtime_heap_alloc_bytes Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')
# TYPE otelcol_process_runtime_heap_alloc_bytes gauge
otelcol_process_runtime_heap_alloc_bytes{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 3.023424e+06
# HELP otelcol_process_runtime_total_alloc_bytes Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')
# TYPE otelcol_process_runtime_total_alloc_bytes counter
otelcol_process_runtime_total_alloc_bytes{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 4.268616e+06
# HELP otelcol_process_runtime_total_sys_memory_bytes Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')
# TYPE otelcol_process_runtime_total_sys_memory_bytes gauge
otelcol_process_runtime_total_sys_memory_bytes{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 1.2274704e+07
# HELP otelcol_process_uptime Uptime of the process
# TYPE otelcol_process_uptime counter
otelcol_process_uptime{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 7.331369
# HELP otelcol_processor_batch_batch_send_size Number of units in the batch
# TYPE otelcol_processor_batch_batch_send_size histogram
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="10"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="25"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="50"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="75"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="100"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="250"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="500"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="750"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="1000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="2000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="3000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="4000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="5000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="6000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="7000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="8000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="9000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="10000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="20000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="30000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="50000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="100000"} 1
otelcol_processor_batch_batch_send_size_bucket{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="+Inf"} 1
otelcol_processor_batch_batch_send_size_sum{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 2
otelcol_processor_batch_batch_send_size_count{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 1
# HELP otelcol_processor_batch_metadata_cardinality Number of distinct metadata value combinations being processed
# TYPE otelcol_processor_batch_metadata_cardinality gauge
otelcol_processor_batch_metadata_cardinality{service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 1
# HELP otelcol_processor_batch_timeout_trigger_send Number of times the batch was sent due to a timeout trigger
# TYPE otelcol_processor_batch_timeout_trigger_send counter
otelcol_processor_batch_timeout_trigger_send{processor="batch",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev"} 1
# HELP otelcol_receiver_accepted_spans Number of spans successfully pushed into the pipeline.
# TYPE otelcol_receiver_accepted_spans counter
otelcol_receiver_accepted_spans{receiver="otlp",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",transport="grpc"} 2
# HELP otelcol_receiver_refused_spans Number of spans that could not be pushed into the pipeline.
# TYPE otelcol_receiver_refused_spans counter
otelcol_receiver_refused_spans{receiver="otlp",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",transport="grpc"} 0
# HELP rpc_server_duration Measures the duration of inbound RPC.
# TYPE rpc_server_duration histogram
rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="0"} 0
rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="5"} 1
rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="10"} 1
rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",service_instance_id="0fd7a1f3-3a39-4825-b8f7-9146185f98fb",service_name="otelcorecol",service_version="0.96.0-dev",le="25"} 1
```

#### OTLP metrics

```
        {
          "Name": "otelcol_processor_batch_metadata_cardinality",
          "Description": "Number of distinct metadata value combinations being processed",
          "Unit": "1",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [],
                "StartTime": "2024-03-15T09:07:20.89055-07:00",
                "Time": "2024-03-15T09:07:56.470161-07:00",
                "Value": 1
              }
            ],
            "Temporality": "CumulativeTemporality",
            "IsMonotonic": false
          }
        }
```

**Review by kristinapathak — APPROVED** [review]
🥳

**Review by mx-psi — APPROVED** [review]

**Comment by codeboten** [comment]
> @codeboten I updated #9767. It'd be great to merge that PR first in order to disable the grps/http instrumentation metrics by default before changing their names

@dmitryax i think that's fine, we can get #9767 merged and released in 0.99.0 and then this change can go out on the following release

**Comment by jaronoff97** [comment]
@codeboten are we waiting on anything for this PR to continue?

**Comment by codeboten** [comment]
@jaronoff97 yeah im just wrapping up the mdatagen work, which will make this PR as easy as updating metadata.yaml files to add the prefix.

**Comment by codeboten** [comment]
@jaronoff97 see:
- https://github.com/open-telemetry/opentelemetry-collector/pull/10316
- https://github.com/open-telemetry/opentelemetry-collector/pull/10318 
- https://github.com/open-telemetry/opentelemetry-collector/pull/10273

**Review by jaronoff97 — COMMENTED** [review]
looks like obsreport may need some fixes too, otherwise this is 🎉 ❤️

**Comment by codeboten** [comment]
This is blocked on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29867, i've not forgotten about this PR

File: exporter/exporterhelper/metadata.yaml
Comment (jpkrohling) [inline exporter/exporterhelper/metadata.yaml:12]: if we are breaking compat by changing the name, shouldn't we also be switching from underscores to dots?

Comment (codeboten) [reply exporter/exporterhelper/metadata.yaml:12]: we're not going to break compat. the `_` is the separator that metrics are emitted with today via prometheus. i'd like to keep that consistent for metrics emitted via OTLP as well

**Comment by codeboten** [comment]
@jpkrohling @jaronoff97 this is ready to move forward now.

Most components in core and contrib using `mdatagen` for metric generation which allows me to make the prefix change in `mdatagen` instead of having to rename all the metrics in `metadata.yaml`.

This forces any metrics listed in `telemetry::metrics` to be prefixed, which I think is what we want for all internal metrics to be prefixed consistently. Note that the documentation generated shows the metric including the prefix.

**Review by jaronoff97 — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
will plan to merge this after Monday's release

**Review by jpkrohling — APPROVED** [review]

File: service/internal/proctelemetry/config.go
Comment (codeboten) [reply service/internal/proctelemetry/config.go:175]: Are you referring to instrumentation library telemetry or telemetry produced by collector component themselves? This was called out as a breaking change in the changelog.

Comment (songy23) [reply service/internal/proctelemetry/config.go:175]: I mean any components that use the [component meter provider](https://github.com/open-telemetry/opentelemetry-collector/blob/6764622672bc3d55ee5eea3ebf0f5ff62f8d7ae8/component/telemetry.go#L25) and export their own metrics together with the collector native metrics.

For example we use the meter provider in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4bac6477d10a371bc44b7f9a937d85996f020584/exporter/datadogexporter/factory.go#L316 and export a bunch of metrics via Prometheus, like `datadog_xyz`. Before this PR they used to have the prefix `otelcol_datadog_xyz` and this PR changed that.

**Pull request merged**
